### PR TITLE
Adding missing swagger specs for automation RP

### LINKED ISSUE #10662: OpenAPI specs for Microsoft.Automation do not match information returned by /providers API
Based on the information returned by the AZ CLI (`az providers list`), resource provider `Microsoft.Automation` should expose resource type `automationAccounts` with the following API versions:
- 2020-01-13-preview
- 2019-06-01
- 2018-06-30
- 2018-01-15
- 2017-05-15-preview
- 2015-10-31
- 2015-01-01-preview

The OpenAPI specs only defines resource type `automationAccounts` with the API version `2015-10-31`

As a result, the auto-generated ARM template schema does not include the correct API versions and there is a conflict between the warnings reported by the `arm-ttk` and VS Code's ARM template schema validation ... as reported here Azure/arm-ttk#184

The OpenAPI spec definition is also not up to date for resource types:
- `automationAccounts/configurations`
- `automationAccounts/webhooks`
- `automationAccounts/softwareUpdateConfigurations`
- `automationAccounts/jobs`

### PR DESCRIPTION
**Pull request by shivamT95-msft**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

Changes:
1. Adding specs for all the missing swagger for S360 Completeness/Correctness KPIs.
3. Fixed most of the validation checks.

**Note: There are no changes from the RP end. Incorrect specs were fixed in the PR. These change will not break any customer scenario**

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [X] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by shivamT95** [comment]
@lirenhe @fengzhou-msft Bump!. We are blocked on swagger changes for enabling support for azure policies. Can we prioritize the reviewing this PR?

**Comment by shivamT95-msft** [comment]
@lirenhe @fengzhou-msft Bump!. We are blocked on swagger changes for enabling support for azure policies. Can we prioritize reviewing this PR?

**Review by JeffreyRichter — APPROVED** [review]

**Comment by qiaozha** [comment]
./azp run

**Comment by shivamT95-msft** [comment]
**Note**
The breaking changes will not break any existing customers since the functionality was always present at the API level and **no change to the service code/ ARM manifest was done**. Summary of breaking changes is as follows:

1. Missing properties in softwareUpdateConfiguration spec. The properties were always present and were being used by portal to configure SUCs.
2. Definition conflict for ScheduleProperties in softwareUpdateConfiguration was resulting in validation failure since we use Schedule properties in schedule.json spec too.
3. Incorrect response type for connection, job specifications: The 409 response type had schema of error response hence it was already covered by the default response. (since we  400+ errors are already in schema). The DELETE operation never returned the JSON of object being deleted. It returned empty response always and examples reflected the same scenario. It will fix the existing documentation and prevent already existing confusion. 
4. PUT runbook always returns a 202 response on success since it is an async API.
5. definitions.json was updated to be in sync with common types in common types folder. It was being flagged in CI validation and is not a breaking change. The default type is object hence it would not break anything and would bring consistency across the public documentation.
6. Removed the API paths that were not present and would return a 404 response.

**Comment by JeffreyRichter** [comment]
Since SDKs are generated from swagger, when swagger changes, you break all the language SDKs produced from the swagger and all the customers and tools (PS/CLI) that use the SDKs.

File: specification/automation/resource-manager/readme.md [hunk 1]
```diff
@@ -156,41 +158,77 @@ input-file:
...
+- Microsoft.Automation/stable/2019-06-01/softwareUpdateConfiguration.json
+- Microsoft.Automation/stable/2019-06-01/hybridRunbookWorkerGroup.json
+- Microsoft.Automation/stable/2019-06-01/jobSchedule.json
+- Microsoft.Automation/stable/2019-06-01/linkedWorkspace.json
+- Microsoft.Automation/stable/2019-06-01/module.json
+- Microsoft.Automation/stable/2019-06-01/operations.json
+- Microsoft.Automation/stable/2019-06-01/schedule.json
+- Microsoft.Automation/stable/2019-06-01/variable.json
+- Microsoft.Automation/stable/2019-06-01/watcher.json
+- Microsoft.Automation/stable/2015-10-31/webhook.json
```
Comment (lirenhe) [inline specification/automation/resource-manager/readme.md:194]: Could you also use the webhook.json with api version of 2019-06-01

File: specification/automation/resource-manager/readme.md [hunk 2]
```diff
@@ -156,41 +158,77 @@ input-file:
...
+- Microsoft.Automation/preview/2020-01-13-preview/connectionType.json
+- Microsoft.Automation/preview/2020-01-13-preview/credential.json
+- Microsoft.Automation/preview/2020-01-13-preview/hybridRunbookWorkerGroup.json
+- Microsoft.Automation/preview/2020-01-13-preview/jobSchedule.json
+- Microsoft.Automation/preview/2020-01-13-preview/linkedWorkspace.json
+- Microsoft.Automation/preview/2020-01-13-preview/module.json
+- Microsoft.Automation/preview/2020-01-13-preview/schedule.json
+- Microsoft.Automation/preview/2020-01-13-preview/variable.json
+- Microsoft.Automation/preview/2020-01-13-preview/watcher.json
+- Microsoft.Automation/stable/2019-06-01/dscConfiguration.json
```
Comment (lirenhe) [inline specification/automation/resource-manager/readme.md:225]: Same above, do you still need to mix api versions in one sdk?

File: specification/automation/resource-manager/readme.md (hunk 1 above)
Comment (shivamT95-msft) [reply specification/automation/resource-manager/readme.md:194]: The api-version  2019-06-01 is not yet supported for webhook resource at the ARM/RP layer. The next stable api-version that we release would have all available operations present.

File: specification/automation/resource-manager/readme.md (hunk 2 above)
Comment (shivamT95-msft) [reply specification/automation/resource-manager/readme.md:225]: Similar reasoning as for  2019-06-01 api-version.

**Review by lirenhe — APPROVED** [review]

**Pull request merged**